### PR TITLE
Paint interop removal

### PR DESCRIPTION
**Pull request by ZehMatt**
Currently in draft as I plan to refactor some things.

### CODE REVIEW

**Comment by ZehMatt** [comment]
I think this will do for now, I don't want to make it too big.

File: src/OpenLoco/src/Paint/Paint.h [hunk 1]
```diff
@@ -122,12 +122,8 @@ namespace OpenLoco::Paint
...
     struct PaintStructBoundBox
     {
-        int16_t x;
-        int16_t y;
-        int16_t z;
-        int16_t xEnd;
-        int16_t yEnd;
-        int16_t zEnd;
+        World::Pos3 mins;
+        World::Pos3 maxs;
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Paint/Paint.h:126]: Guess `min` and `max` are reserved names, right? Due to the `s`, this reads as plural, though.

File: src/OpenLoco/src/Paint/Paint.h [hunk 2]
```diff
@@ -99,38 +101,29 @@ namespace OpenLoco::Paint
...
+        uint16_t colour;
     };
-    assert_struct_size(PaintStringStruct, 0x20);
 
     struct PaintStructBoundBox
     {
-        int16_t x;    // 0x08
-        int16_t y;    // 0x0A
-        int16_t z;    // 0x0C
-        int16_t zEnd; // 0x0E NOTE: This order is important!
```
Comment (LeftofZen) [inline src/OpenLoco/src/Paint/Paint.h:131]: is this order still important? or does your refactor change that?

Comment (AaronVanGeffen) [reply src/OpenLoco/src/Paint/Paint.h:131]: Now that interop has been removed, the order is no longer important.

File: src/OpenLoco/src/Paint/Paint.h (hunk 1 above)
Comment (ZehMatt) [reply src/OpenLoco/src/Paint/Paint.h:126]: Yeah min/max are often macros, I just went with what Valve has:
image

File: src/OpenLoco/src/Paint/Paint.cpp [hunk 3]
```diff
@@ -965,23 +965,10 @@ namespace OpenLoco::Paint
...
-                    imagePos.x &= 0xFFFC;
-                    imagePos.y &= 0xFFFC;
-                    break;
-                case 3:
-                    imagePos.x &= 0xFFF8;
-                    imagePos.y &= 0xFFF8;
-                    break;
-            }
+            // Align the position to 0, 2, 4, 8 based on the zoom level.
+            const auto mask = static_cast<int16_t>(~(1U << rt.zoomLevel) + 1);
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/Paint/Paint.cpp:969]: I feel `mask = 0xFFFFU << zoomLevel` it's easier to understand

File: src/Core/include/OpenLoco/Core/Numerics.hpp [hunk 4]
```diff
@@ -27,4 +27,16 @@ namespace OpenLoco::Numerics
     {
         return ((val >> rotation) | (val << (4 - rotation))) & 0xF;
     }
+
+    template<typename T>
+    constexpr T alignDown(T value, size_t alignment)
+    {
+        return value & ~(alignment - 1);
```
Comment (duncanspumpkin) [inline src/Core/include/OpenLoco/Core/Numerics.hpp:34]: I find this much harder to understand than `& ~0 <<alignment`

Comment (ZehMatt) [reply src/Core/include/OpenLoco/Core/Numerics.hpp:34]: It is however less efficient in terms of compiled output code.
```
alignDown32_PR(unsigned int, unsigned long):
        mov     eax, esi
        neg     eax
        and     eax, edi
        ret
alignDown32_Duncan(unsigned int, unsigned long):
        mov     ecx, esi
        mov     eax, -1
        sal     eax, cl
        and     eax, edi
        ret
```

**Review by AaronVanGeffen — APPROVED** [review]
Nice work. Very happy to see us moving towards a completely independent codebase.

**Comment by duncanspumpkin** [comment]
Have you checked that vehicle routing still works? Its the only thing that isn't implemented that uses viewport interaction.

**Comment by ZehMatt** [comment]
> Have you checked that vehicle routing selection still works? Its the only thing that isn't implemented that uses viewport interaction.

If you mean the Express option from the routing tab then yes that seems to work when clicking on stations.

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**